core/pkg/artifacts/downloader.go

```diff
@@ -185,6 +177,7 @@ func (ad *ArtifactDownloader) downloadFiles(artifactID string, manifest Manifest
 					continue
 				}
 				numDone++
+				go ad.FileCache.AddFileAndCheckDigest(result.Task.Path, manifest.Contents[result.Name].Digest)
```

Comment:
Part of the assumption here is that caching is always optional--we're not waiting to check the error result of the call because if it fails that's not a critical issue. We should make sure it shows up in the debug log, but ultimately the cache is just there for performance reasons, and we don't (and can't) rely on any given item being in the cache at any point. So I'm going to say yes, if the core process finishes or is killed then it's fine that we didn't write to cache.

If you're saying that goroutines called from other goroutines die when their parent ends, even if the root process is still alive, then that is a problem, but that's not how I understood goroutines to work. We do expect this function to often return before the goroutine finishes, but I expect that the core process will generally still be alive.

Reply:
Sounds good!

> If you're saying that goroutines called from other goroutines die when their parent ends, even if the root process is still alive, then that is a problem

Nope, just when the root process exits. So it sounds like it should work fine.